localstack/services/sqs/utils.py

```diff
@@ -34,6 +34,37 @@ def is_sqs_queue_url(url: str) -> bool:
     )
 
 
+def guess_endpoint_strategy_and_host(
+    host: str,
+) -> Tuple[Literal["standard", "domain", "path"], str]:
+    """
+    This method is used for the dynamic endpoint strategy. It heuristically determines a tuple where the first
+    element is the endpoint strategy, and the second is the part of the host after the endpoint prefix and region.
+    For instance:
+
+      * ``sqs.us-east-1.localhost.localstack.cloud`` -> ``standard, localhost.localstack.cloud``
+      * ``queue.localhost.localstack.cloud:4566`` -> ``domain, localhost.localstack.cloud:4566``
+      * ``us-east-2.queue.amazonaws.com`` -> ``domain, amazonaws.com``
+      * ``localhost:4566`` -> ``path, localhost:443``
+      * ``amazonaws.com`` -> ``path, amazonaws.com``
+
+    :param host: the original host in the request
+    :return: endpoint strategy, host segment
+    """
+    components = host.split(".")
+
+    if host.startswith("sqs."):
+        return "standard", ".".join(components[2:])
+
+    if host.startswith("queue."):
+        return "domain", ".".join(components[1:])
+
+    if len(components) > 2 and components[1] == "queue":
+        return "domain", ".".join(components[2:])
```

Comment:
I am probably forgetting something, but why len > 2? The examples listed above would suggest at least len > 3 (`<region>.queue.(amazonaws | localhost.localstack).com`). Or is it to avoid a IndexError? but then > 1 is enough :)

Reply:
perhaps to allow `us-west-1.queue.localhost` but not `us-west-1.queue`. although it's not clear to me why anyone would name their host `queue`, but not unthinkable :D